instrumentation/kotlin-coroutines/library/src/main/kotlin/io/opentelemetry/instrumentation/kotlincoroutines/ContextExtensions.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.kotlincoroutines
+
+import io.opentelemetry.api.trace.Span
+import io.opentelemetry.context.Context
+import io.opentelemetry.javaagent.instrumentation.api.Java8BytecodeBridge
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Returns a [CoroutineContext] which will make this [Context] current when resuming a coroutine
+ * and restores the previous [Context] on suspension.
+ */
+fun Context.asContextElement(): CoroutineContext {
```

Comment:
By the way, I'm wondering whether these should be in the SDK instead of instrumentation. Mainly since I was imagining adding to the javadoc for `makeCurrent` to not use it in Kotlin coroutines and use this instead. But that doesn't make as much sense when crossing repos.

Reply:
language-specific extensions to the API, e.g. `opentelemetry-extension-kotlin`? I like it

Reply:
Sent https://github.com/open-telemetry/opentelemetry-java/pull/2066 for the library aspect of this PR

Reply:
I agree - this is a great approach.